src/common/interfaces/IGateway.sol

```diff
@@ -38,29 +38,35 @@ interface IGateway is IMessageHandler, IMessageSender, IGatewayHandler {
         uint64 activeSessionId;
     }
 
-    struct Message {
+    struct InboundBatch {
         /// @dev Counts are stored as integers (instead of boolean values) to accommodate duplicate
         ///      messages (e.g. two investments from the same user with the same amount) being
         ///      processed in parallel. The entire struct is packed in a single bytes32 slot.
         ///      Max uint16 = 65,535 so at most 65,535 duplicate messages can be processed in parallel.
         uint16[MAX_ADAPTER_COUNT] votes;
         /// @notice Each time adapters are updated, a new session starts which invalidates old votes
         uint64 sessionId;
-        bytes pendingMessage;
+        bytes pendingBatch;
     }
 
     // --- Events ---
-    event ProcessMessage(uint16 centrifugeId, bytes message, IAdapter adapter);
+    event ProcessBatch(uint16 centrifugeId, bytes batch, IAdapter adapter);
     event ProcessProof(uint16 centrifugeId, bytes32 messageHash, IAdapter adapter);
-    event ExecuteMessage(uint16 centrifugeId, bytes message, IAdapter adapter);
-    event SendMessage(bytes message);
+    event ExecuteMessage(uint16 centrifugeId, bytes message);
+    event FailMessage(uint16 centrifugeId, bytes message, bytes error);
+    event SendBatch(uint16 centrifugeId, bytes batch, IAdapter adapter);
+    event SendProof(uint16 centrifugeId, bytes proof, IAdapter adapter);
+    event PrepareMessage(uint16 centrifugeId, PoolId poolId, bytes message);
```

Comment:
I think the events `ExecuteMessage`, `FailMessage` and `PrepareMessage` does not require a `centrifugeId`, parameter because they are always attached to a `ProcessBatch`/`SendBatch` message that already contains that information.

If you agree, I will remove those parameters

Reply:
Indexer looks at individual events rather than groups of events so would keep it.